packages/compass-saved-aggregations-queries/src/stores/aggregations-queries-items.ts

```diff
@@ -52,20 +54,24 @@ const INITIAL_STATE: State = {
 const favoriteQueryStorage = new FavoriteQueryStorage();
 const pipelineStorage = new PipelineStorage();
 
-const reducer: Reducer<State, Actions> = (state = INITIAL_STATE, action) => {
-  if (action.type === ActionTypes.ITEMS_FETCHED) {
-    return {
-      ...state,
-      items: action.payload,
-      loading: false,
-    };
-  }
-  if (action.type === ActionTypes.ITEM_DELETED) {
-    const newItems = state.items.filter((item) => item.id !== action.id);
-    return {
-      ...state,
-      items: newItems,
-    };
+const reducer: Reducer<State, Actions | DeleteItemActions> = (
+  state = INITIAL_STATE,
+  action
+) => {
+  switch (action.type) {
+    case ActionTypes.ITEMS_FETCHED:
+      return {
+        ...state,
+        items: action.payload,
+        loading: false,
+      };
+    case DeleteItemActionTypes.DeleteItemConfirm: {
```

Comment:
This aligns the implementation with [many reducers respond to one action](https://redux.js.org/style-guide/style-guide#allow-many-reducers-to-respond-to-the-same-action) recommendation